maat-court-data-api/src/main/java/gov/uk/courtdata/dces/service/ConcorContributionsService.java

```diff
@@ -82,24 +77,20 @@ public Integer createContributionAndUpdateConcorStatus(CreateContributionFileReq
 
     @Transactional(rollbackFor =  MAATCourtDataException.class)
     @NotNull
-    public Integer logContributionProcessed(LogContributionProcessedRequest request){
+    public Integer logContributionProcessed(LogContributionProcessedRequest request) {
         ConcorContributionsEntity concorEntity = concorRepository.findById(request.getConcorId())
                 .orElseThrow(() -> new RequestedObjectNotFoundException("concor_contribution could not be found by id"));
         log.info("Contribution found: {}", concorEntity.getId());
-        if (!debtCollectionService.updateContributionFileReceivedCount(concorEntity.getContribFileId())) {
-            throw new NoSuchElementException("No associated contribution_file found for concur_contribution");
-        }
-        if(!StringUtils.isEmpty(request.getErrorText()) ){
+        if (!StringUtils.isEmpty(request.getErrorText())) {
             saveErrorMessage(request, concorEntity);
+        } else if (!debtCollectionService.updateContributionFileReceivedCount(concorEntity.getContribFileId())) {
+            throw new NoSuchElementException("No associated contribution_file found for concur_contribution");
```

Comment:
I would include the `ContribFileId` in the exception message to aid debugging.

Reply:
Will do - was being overcautious: an int can't cause XSS, and this service is consumed by another service (not a browser) anyway!